src/connection-manager/agreement.ts

```diff
@@ -169,13 +249,16 @@ class Agreement {
     if (!hashes.has(validation.ledger_hash)) {
       hashes.set(validation.ledger_hash, Date.now())
       this.validationsByPublicKey.set(signing_key, hashes)
+      const release_version = decodeReleaseVersion(validation.server_version)
       const validator = {
         master_key: validation.master_key,
         signing_key,
         ledger_hash: validation.ledger_hash,
         current_index: Number(validation.ledger_index),
         partial: !validation.full,
         last_ledger_time: new Date(),
+        ...(preceedFlagLedger(validation.ledger_index) &&
+          release_version && { version: release_version }),
```

Comment:
```suggestion
        version: release_version,
```

Reply:
I did this so that version will not be updated if there's ledger with no validation. With the change version: undefined will replace an actual version. Any ideas?

Reply:
Made some changes. Let me know what you think. Thanks Mayukha!